code/mob/living/critter/ai/sawflyai.dm

```diff
@@ -0,0 +1,175 @@
+//For the sawfly critter itself, check mob/living/critter/sawfly.dm
+//For misc things, check sawflymisc.dm in code/obj.sawflymisc
+
+/datum/aiHolder/sawfly
+
+	New()
+		..()
+		default_task = get_instance(/datum/aiTask/prioritizer/sawfly, list(src))
+
+/datum/aiTask/prioritizer/sawfly
+	name = "base thinking (should never see this)"
+
+/datum/aiTask/prioritizer/sawfly/New()
+	..()
+	// populate the list of tasks
+	transition_tasks += holder.get_instance(/datum/aiTask/timed/wander, list(holder, src))
+	transition_tasks += holder.get_instance(/datum/aiTask/timed/targeted/sawfly_attack, list(holder, src))
+	transition_tasks += holder.get_instance(/datum/aiTask/sequence/goalbased/sawfly_chase_n_stab, list(holder, src))
+
+/datum/aiTask/prioritizer/sawfly/on_tick()
+	if(isdead(holder.owner))
+		holder.enabled = FALSE
+		walk(holder.owner, 0)
+
+/datum/aiTask/prioritizer/sawfly/on_reset()
+	..()
+	walk(holder.owner, 0)
+
+// Custom behaviour starts here
+
+// Attack code - looks for adjacent targets, tries to stab them
+/datum/aiTask/timed/targeted/sawfly_attack
+	var/found_path = null
+	name = "attacking"
+	minimum_task_ticks = 10
+	maximum_task_ticks = 25
+	var/weight = 10
+	target_range = 1
+	var/list/dummy_params = list("icon-x" = 16, "icon-y" = 16)
+
+/datum/aiTask/timed/targeted/sawfly_attack/evaluate()
+	return weight * score_target(get_best_target(get_targets()))
+
+/datum/aiTask/timed/targeted/sawfly_attack/on_tick()
+	var/mob/living/critter/robotic/sawfly/owncritter = holder.owner
+	if(prob(5)) owncritter.communalbeep()
+	walk(owncritter, 0)
+	if(!holder.target)
+		holder.target = get_best_target(get_targets())
+	if(holder.target)
+		var/atom/T = holder.target
+		// if target is dead
+		// fetch a new one if we can
+		if(isliving(T))
+			var/mob/living/M = T
+			if(M.health < -50)
+				holder.target = get_best_target(get_targets())
+			if(istype(T, /mob/living/critter/robotic/sawfly))
+				holder.target = get_best_target(get_targets())
+		if(!holder.target) //we lost target, re-evaluate tasks
+			holder.interrupt()
+			return
+
+		var/dist = get_dist(owncritter, holder.target)
+		if(dist > target_range)
+			if(!src.found_path)
+				src.found_path = get_path_to(holder.owner, holder.target, 18, 0)
```

Comment:
why 18?

Reply:
It's slightly larger than the size of the screen and allows them to chase you beyond your view. It can be toned down but I felt it was a high enough number to make them pursue thoroughly without seeking too aggressively

Reply:
seems reasonable to me - it might be worth putting a comment to that effect there, just to clarify. Arbitrary numbers without justification should be avoided if possible